File: Settings/src/org/nameless/settings/display/ForceFullscreenDisplayFragment.java
/*
 * Copyright (C) 2024 The Nameless-AOSP Project
 * SPDX-License-Identifier: Apache-2.0
 */

package org.nameless.settings.display;

import androidx.appcompat.app.AlertDialog;

import com.android.settings.R;

import java.util.List;

import org.nameless.custom.preference.SwitchPreferenceCompat;
import org.nameless.settings.fragment.PerAppSwitchConfigFragment;

public class ForceFullscreenDisplayFragment extends PerAppSwitchConfigFragment {

    @Override
    protected int getAllowedSystemAppListResId() {
        return R.array.config_fullscreenDisplayAllowedSystemApps;
    }

    @Override
    protected int getPreferenceScreenResId() {
        return R.xml.force_fullscreen_display;
    }

    @Override
    protected int getTopInfoResId() {
        return R.string.app_force_fullscreen_info;
    }

    @Override
    protected boolean isChecked(String packageName, int uid) {
        return mPackageManager.isForceFull(packageName);
    }

    @Override
    protected boolean onSetChecked(SwitchPreferenceCompat pref, String packageName, int uid, boolean checked) {
        new AlertDialog.Builder(mContext)
                .setTitle(mContext.getText(R.string.switch_fullscreen_display_warn_title))
                .setMessage(mContext.getText(R.string.switch_fullscreen_display_warn_content))
                .setPositiveButton(R.string.okay, (dialog, which) -> {
                    mPackageManager.setForceFull(packageName, checked);
                    pref.setChecked(checked);
                })
                .setNegativeButton(R.string.cancel, null)
                .show();
        return false;
    }

    @Override
    protected void onCheckedListUpdated(List<String> pkgList) {}
}

File: Settings/src/org/sun/settings/display/ForceFullscreenDisplayFragment.kt
/*
 * Copyright (C) 2025 The Nameless-CLO Project
 * SPDX-License-Identifier: Apache-2.0
 */

package org.sun.settings.display

import androidx.appcompat.app.AlertDialog

import com.android.settings.R

import org.sun.custom.preference.SwitchPreferenceCompat
import org.sun.settings.fragment.PerAppSwitchConfigFragment

class ForceFullscreenDisplayFragment : PerAppSwitchConfigFragment() {

    override fun getAllowedSystemAppListResId() = R.array.config_fullscreenDisplayAllowedSystemApps

    override fun getBlacklistAppListResId() = com.android.internal.R.array.config_forceFullBlacklistApps

    override fun getTitleResId() = R.string.app_force_fullscreen_title

    override fun getTopInfoResId() = R.string.app_force_fullscreen_info

    override fun isChecked(packageName: String, uid: Int): Boolean {
        return packageManager.isForceFull(packageName)
    }

    override fun onSetChecked(pref: SwitchPreferenceCompat, packageName: String, uid: Int, checked: Boolean): Boolean {
        AlertDialog.Builder(requireContext()).apply {
            setTitle(requireContext().getText(R.string.switch_fullscreen_display_warn_title))
            setMessage(requireContext().getText(R.string.switch_fullscreen_display_warn_content))
            setPositiveButton(R.string.okay, { _, _ ->
                packageManager.setForceFull(packageName, checked)
                pref.isChecked = checked
            })
            setNegativeButton(R.string.cancel, null)
        }.show()
        return false
    }

    override fun onCheckedListUpdated(pkgList: List<String>) {}

    override fun getSelectAllRunnable(): Runnable? {
        return Runnable {
            AlertDialog.Builder(requireContext()).apply {
                setTitle(requireContext().getText(R.string.switch_fullscreen_display_warn_title))
                setMessage(requireContext().getText(R.string.switch_fullscreen_display_warn_content))
                setPositiveButton(R.string.okay, { _, _ ->
                    super.getSelectAllRunnable()?.run()
                    allAppData.forEach {
                        packageManager.setForceFull(it.packageName, true)
                    }
                })
                setNegativeButton(R.string.cancel, null)
            }.show()
        }
    }

    override fun getDeselectAllRunnable(): Runnable? {
        return Runnable {
            AlertDialog.Builder(requireContext()).apply {
                setTitle(requireContext().getText(R.string.switch_fullscreen_display_warn_title))
                setMessage(requireContext().getText(R.string.switch_fullscreen_display_warn_content))
                setPositiveButton(R.string.okay, { _, _ ->
                    super.getDeselectAllRunnable()?.run()
                    allAppData.forEach {
                        packageManager.setForceFull(it.packageName, false)
                    }
                })
                setNegativeButton(R.string.cancel, null)
            }.show()
        }
    }

    override fun getResetRunnable(): Runnable? {
        return Runnable {
            AlertDialog.Builder(requireContext()).apply {
                setTitle(requireContext().getText(R.string.switch_fullscreen_display_warn_title))
                setMessage(requireContext().getText(R.string.switch_fullscreen_display_warn_content))
                setPositiveButton(R.string.okay, { _, _ ->
                    super.getSelectAllRunnable()?.run()
                    allAppData.forEach {
                        packageManager.setForceFull(it.packageName, true)
                    }
                    showResetSuccessToast()
                })
                setNegativeButton(R.string.cancel, null)
            }.show()
        }
    }
}
